keps/sig-node/5067-pod-generation/README.md

```diff
@@ -0,0 +1,972 @@
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+One of the motivations for this KEP comes from the existing ResizeStatus field. 
+In its original implementation, the ResizeStatus field was written to by both 
+the API server and the Kubelet, creating a race condition on updates. Removing 
+the Proposed state makes the Kubelet the only writer to that field, but leaves a
+gap in knowing whether the latest resize has been acknowledged by the Kubelet. 
+The changes proposed in this KEP resolve this gap.
+
+The ResizeStatus is used as an example here, but in practice this issue can be 
+generally found in any type of pod update.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+* Provide a general solution for the pod status to express which pod update is 
+currently being reflected.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+* Address more specific issues with any particular pod status field(s). 
+* Expand the set of mutable fields.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### API Changes
+
+#### Generation and ObservedGeneration
+
+The `metadata.generation` field is currently unused on pods, but we can start 
+leveraging it to help track which pod state is currently being reflected in the 
+pod status. [`metadata.generation`](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#metadata) is documented as "a sequence
+number representing a specific generation of the desired state. Set by the 
+system and monotonically increasing, per-resource." For consistency, pod 
+`metadata.generation` will be incremented whenever the pod has changes that the kubelet
+needs to actuate. 
+
+A new field `status.observedGeneration` field will be added to the pod status.
+Kubelet will set this to communicate which pod state is being expressed in the 
+current pod status. This is analogous to the `status.observedGeneration` that exists
+in other resources' statuses such as [StatefulSets](https://github.com/kubernetes/api/blob/9e7d345b161c12a3056efa88e285b3ef68450c54/apps/v1/types.go#L274).
+
+Likewise, a new `observedGeneration` field will be added to the pod's 
```

Comment:
Both of the new fields should be optional, btw.

Reply:
added the word 'optional' to the body of the kep